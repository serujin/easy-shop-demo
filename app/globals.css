@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
