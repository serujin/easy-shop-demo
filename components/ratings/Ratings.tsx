import React from "react";
import styles from "./Ratings.module.css";
import StarDisplay from "./StarDisplay";

type RatingsProps = {
  titleStyle?: "huge" | "normal";
  title: string;
  rating: number;
  reviews: number;
};

const createRatingsText = (reviews: number) => `${reviews} reviews`;

export default function Ratings(props: RatingsProps) {
  const { title, titleStyle = "normal", rating, reviews } = props;
  const ratingsText = createRatingsText(reviews);
  return (
    <div className={styles.container}>
      <div className={styles.titleContainer}>
        {titleStyle === "huge" ? <h1>{title}</h1> : <p>{title}</p>}
      </div>
      <div className={styles.reviewsContainer}>
        <StarDisplay rating={rating} maxRating={5} />
        <p>{ratingsText}</p>
      </div>
    </div>
  );
}
