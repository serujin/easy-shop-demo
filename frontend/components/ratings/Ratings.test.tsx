import React from 'react'
import { render } from '@testing-library/react'
import Ratings from './Ratings'
import { describe, expect, it } from '@jest/globals'

describe('Ratings component', () => {
  describe('when receiving huge as a titleStyle', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Ratings rating={5} reviews={20000} title="title" titleStyle="huge" />,
      )

      expect(container).toMatchSnapshot()
    })
  })
  describe('when not receiving any titleStyle', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Ratings rating={5} reviews={20000} title="title" />,
      )

      expect(container).toMatchSnapshot()
    })
  })
})
