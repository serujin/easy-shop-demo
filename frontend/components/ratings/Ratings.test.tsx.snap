// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings component when not receiving any titleStyle should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="titleContainer"
    >
      <p
        class="normal"
      >
        title
      </p>
    </div>
    <div
      class="reviewsContainer normal"
    >
      <div
        class="starsContainer"
      >
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
      </div>
      <p>
        20,000 reviews
      </p>
    </div>
  </div>
</div>
`;

exports[`Ratings component when receiving huge as a titleStyle should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="titleContainer"
    >
      <p
        class="huge"
      >
        title
      </p>
    </div>
    <div
      class="reviewsContainer huge"
    >
      <div
        class="starsContainer"
      >
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
        <img
          alt="Full star"
          class="starItem"
          data-nimg="1"
          decoding="async"
          height="17"
          loading="lazy"
          src="/ratings/full-star.svg"
          style="color: transparent;"
          width="17"
        />
      </div>
      <p>
        20,000 reviews
      </p>
    </div>
  </div>
</div>
`;
